packages/blade/src/components/Typography/BaseText/BaseText.web.tsx

```diff
@@ -0,0 +1,69 @@
+import type { ReactElement } from 'react';
+import styled from 'styled-components';
+import getBaseTextStyles from './getBaseTextStyles';
+import type { BaseTextProps, StyledBaseTextProps } from './BaseTextTypes';
+import { makeAccessible } from '~utils';
+
+const StyledBaseText = styled.div<StyledBaseTextProps>(
+  ({
+    color,
+    fontFamily,
+    fontSize,
+    fontWeight,
+    fontStyle,
+    textDecorationLine,
+    lineHeight,
+    textAlign,
+    ...props
+  }) =>
+    getBaseTextStyles({
+      color,
+      fontFamily,
+      fontSize,
+      fontWeight,
+      fontStyle,
+      textDecorationLine,
+      lineHeight,
+      textAlign,
+      theme: props.theme,
+    }),
+);
+
+export const BaseText = ({
+  id,
+  color,
+  fontFamily,
+  fontSize,
+  fontWeight,
+  fontStyle,
+  textDecorationLine,
+  lineHeight,
+  as,
+  textAlign,
+  children,
+  truncateAfterLines,
+  className,
+  style,
+  accessibilityProps = {},
+}: BaseTextProps): ReactElement => {
+  return (
+    <StyledBaseText
+      color={color}
+      fontFamily={fontFamily}
+      fontSize={fontSize}
+      fontWeight={fontWeight}
+      fontStyle={fontStyle}
+      textDecorationLine={textDecorationLine}
+      lineHeight={lineHeight}
+      as={as}
+      textAlign={textAlign}
+      numberOfLines={truncateAfterLines}
+      className={className}
+      style={style}
```

Comment:
Non blocker: We might want to consider removing `style` prop here. Earlier we (I) introduced this because there was no mechanism of adding padding between icon & text of Button. With the Box component, this need is resolved and its usage is also removed with [this PR](https://github.com/razorpay/blade/pull/543). 

Exposing style directly is a bit risky and we should be exposing this in only extreme situations

Reply:
Probably we can do it in another PR?